test/gtest/graphapi_capi_mha_backward.cpp

```diff
@@ -0,0 +1,430 @@
+            softmax,
+            mDesc,
+            zInvDesc,
+            q.mDescale,
+            k.mDescale,
+            v.mDescale,
+            sDescale,
+            sScale,
+            oScale,
+            m_bernulliProbability,
+            GetTensor<int64_t>(m_realTensorMap[miopenTensorMhaDropoutSeed]->m_tensorVariant)
+                .data.front(),
+            GetTensor<int64_t>(m_realTensorMap[miopenTensorMhaDropoutOffset]->m_tensorVariant)
+                .data.front(),
+            amaxS,
+            amaxO,
+            oDesc);
+
+        auto dO = test::cpu::GenScaledTensorBackward<dO_T>(m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDO, m_testN, m_testH, m_testS, m_testD)
+            .InitAndWriteToGPU(handle, std::move(dO.mTensor));
+
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaO, m_testN, m_testH, m_testS, m_testD)
+            .InitAndWriteToGPU(handle, std::move(oDesc));
+
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaM, m_testN, m_testH, m_testS, 1)
+            .InitAndWriteToGPU(handle, std::move(mDesc));
+
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaZInv, m_testN, m_testH, m_testS, 1)
+            .InitAndWriteToGPU(handle, std::move(zInvDesc));
+
+        float dsScale   = 1.0f;
+        float dsDescale = 1.0f;
+
+        float dqScale = 1.0f;
+        float dkScale = 1.0f;
+        float dvScale = 1.0f;
+
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDescaleO).InitAndWriteToGPU(handle, oDescale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDescaleDO)
+            .InitAndWriteToGPU(handle, dO.mDescale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDescaleDS)
+            .InitAndWriteToGPU(handle, dsDescale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDS).InitAndWriteToGPU(handle, dsScale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDQ).InitAndWriteToGPU(handle, dqScale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDK).InitAndWriteToGPU(handle, dkScale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDV).InitAndWriteToGPU(handle, dvScale);
+
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDQ, m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDK, m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDV, m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDQ);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDK);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDV);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDS);
+
+        dQDescRef = tensor<T>{m_testN, m_testH, m_testS, m_testD};
+        dKDescRef = tensor<T>{m_testN, m_testH, m_testS, m_testD};
+        dVDescRef = tensor<T>{m_testN, m_testH, m_testS, m_testD};
+
+        test::cpu::MultiHeadAttentionBackwardDataf8(
```

Comment:
function name suggests that you'll be creating tensors, but you're also calling cpu reference.

Reply:
Yeah its convenient  to do it here, because this call uses a lot from previous calculations. Probably worth to consider slightly rename it